.github/workflows/github_pages_preview.yml

```diff
@@ -0,0 +1,79 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Preview
+
+on:
+  pull_request:
+    paths-ignore:
+      - '.github/labels.yml'
+      - '.github/workflows/labels.yaml'
+      - '.gitignore'
+      - '.gitmodules'
+      - 'CONTRIBUTING.md'
+      - 'LICENSE'
+      - 'README.md'
+
+jobs:
+  deploy:
+    runs-on: ubuntu-20.04
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.event.number || github.ref }}
```

Comment:
The contents of github.event depend on how the workflow is triggered. I've included the fallback here because it felt safer to create a race around a commit than around something more broadly undefined.

Not sure what matching the directory name gets us, as the unique value is the PR number used here.

Reply:
Maybe I'm misunderstanding, but my understanding is that we are essentially worried about a race condition where multiple triggers happen relatively close together. The main "critical area" is when they commit/write their code to the folder. 

So what we really care about is when looking for conflicts is _where_ they write the code too. If we are using a concurrency group on `github.event.number || github.ref`, we're still pusing to `./previews/PR-${{ github.event.number }}`. So if `github.event.number` is empty but `github.ref` is different we'll still have a conflict - they'll push to `./previews/PR-`.

Reply:
Ok, we have two paths:

* Remove the fallback, add a comment above that the workflow only supports pull requests because we depend on `github.event.number`
* Add the fallback everywhere, if it's triggered we deploy to PR-sha. Could maybe do more variable magic to clean that path up

I vote for the first option, simplicity.

Reply:
SGTM